forem/forem
# Feature: Runtime Content Filter

## What type of PR is this? (check all applicable)


## Description

Implements the Runtime Content Filter feature described in [RFC 27](https://github.com/forem/rfcs/pull/27)

## Related Tickets & Documents

https://github.com/forem/rfcs/pull/27

## QA Instructions, Screenshots, Recordings

1. Boot app locally
2. Enable the `runtime_banner` Feature Flag
   - I like to use the console for this so `Flipper.enable(:runtime_banner)` would do the trick
4. Open any article (any page really) on your desktop browser 
   - You **shouldn't be able to see** a floating dark background banner with the text "Open with Forem app"
   - Check the `body` element and you should see `data-runtime` attribute that matches your environment. In macOS I get `Browser-macOS`
6. Open any article (any page really) using your local mobile device (DevTools mobile view won't work)
   - You **should be able to see** a floating dark background banner with the text "Open with Forem app"

### UI accessibility concerns?

No accessibility concerns for this PR. 

A follow up PR that will include the Runtime Banner (and also removing the Feature Flag) will require to be accessible though.

## Added tests?


## [Forem core team only] How will this change be communicated?

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)

The above ^ is a WIP

## [optional] What gif best describes this PR or how it makes you feel?

![what do you see?](https://media.giphy.com/media/3o6nVaI0XQqAemNmaA/giphy.gif)
